src/pip/_internal/utils/subprocess.py

```diff
@@ -198,7 +196,9 @@ def call_subprocess(
     all_output = []
     while True:
         # The "line" value is a unicode string in Python 2.
-        line = console_to_str(proc.stdout.readline())
+        line = None
+        if proc.stdout:
+            line = console_to_str(proc.stdout.readline())
```

Comment:
```suggestion
        line = console_to_str(proc.stdout.readline())
```

Reply:
Fixed in https://github.com/pypa/pip/pull/8322/commits/a847ddab25a2502dc9f0402347c83c8d9955c022